### ISSUE TITLE
The discussion progress indicator now seems to go away too quickly

### ISSUE THREAD

**Issue by crspybits** [crspybits/SharedImages#82]
On a first download of all images/discussion threads, the progress indicator goes away but the spinner is still spinner and images still seem to be downloading. For maybe up to 10 images?

**Comment by crspybits** [comment]
This *may* be due to another issue. Not what I was thinking of. I just tried to reproduce this and this is what happened:

1) I triggered the download of all images for a "freshly" installed app.
2) While the download was occurring, I added a new comment to one of Rod's images-- from a different device.
3) And, of course, that caused and increment in the master version. 
4) Which caused the current download to fail, and the download to restart.
5) And, somehow, this caused the progress indicator to go away. And not be replaced by another. [[This is an error]].

My hypothesis is: If I leave the fresh download of a new app instance to proceed without an update to the master version, the progress indicator should remain present.
